frappe/desk/doctype/dashboard_chart/dashboard_chart.js

```diff
@@ -9,6 +9,10 @@ frappe.ui.form.on('Dashboard Chart', {
 		frm.add_fetch('source', 'timeseries', 'timeseries');
 	},
 
+	onload: function(frm) {
+		frm.set_df_property('parent_document_type', 'hidden', true);
+	},
```

Comment:
This will hide the field whenever the user navigates to the document even when the conditions to show are met.